src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -369,18 +369,14 @@ export function IndexPatternProvider(Private, $http, config, kbnIndex, Promise,
           return confirmModalPromise(confirmMessage, { confirmButtonText: 'Edit existing pattern' })
             .then(() => {
               kbnUrl.change('/management/kibana/indices/{{id}}', { id: duplicate.id });
-              return true;
+              return Promise.reject();
             })
-            .catch(() => {
-              throw new IndexPatternAlreadyExists(this.title);
-            });
+            .catch(() => Promise.reject());
```

Comment:
Is there a reason to return a Promise rejection here or on line 372? If I'm following this logic correctly, we just want to tell the call site of this function "Hey, stop what you're doing, because you can't actually create an index pattern."

In which case, the Promise rejection has nothing to do with the confirm modal, and should probably happen outside of it:

```javascript
          confirmModalPromise(confirmMessage, { confirmButtonText: 'Edit existing pattern' })
            .then(() => {
              kbnUrl.change('/management/kibana/indices/{{id}}', { id: duplicate.id });
            });

          // Abort index pattern creation.
          return Promise.reject();
```